web/server/codechecker_server/api/report_server.py

```diff
@@ -1049,6 +1050,84 @@ def get_cleanup_plan_report_hashes(
     return cleanup_plan_hashes
 
 
+def sort_review_statuses_query(
+    query,
+    sort_mode: ReviewStatusRuleSortMode,
+    report_count_label
+):
+    """
+    Sort review status rule query by the given sort mode.
+    """
+    # Sort by rule date by default.
+    if not sort_mode:
+        return query.order_by(desc(ReviewStatus.date))
+
+    order_type_map = {Order.ASC: asc, Order.DESC: desc}
+    order_type = order_type_map.get(sort_mode.ord)
+    if sort_mode.type == ReviewStatusRuleSortType.REPORT_HASH:
+        query = query.order_by(order_type(ReviewStatus.bug_hash))
+    elif sort_mode.type == ReviewStatusRuleSortType.STATUS:
+        query = query.order_by(order_type(ReviewStatus.status))
+    elif sort_mode.type == ReviewStatusRuleSortType.AUTHOR:
+        query = query.order_by(order_type(ReviewStatus.author))
+    elif sort_mode.type == ReviewStatusRuleSortType.DATE:
+        query = query.order_by(order_type(ReviewStatus.date))
+    elif sort_mode.type == ReviewStatusRuleSortType.ASSOCIATED_REPORTS_COUNT:
+        query = query.order_by(order_type(report_count_label))
+
+    return query
+
+
+def process_rs_rule_filter(
+    query,
+    rule_filter: ReviewStatusRuleFilter
+):
+    """ Process review status rule filter. """
+    if rule_filter:
```

Comment:
It can be (you can call the `getReviewStatusRules` API function where the `rule_filter` is `None`), so I fixed the type of this parameter to `Optional`.

Reply:
I feel this unnecessarily defensive, because in this regard we should check every function parameter in Python, since any parameter can be `None`. Though it doesn't make any harm, so it can be merged.